### PR TITLE
Added inspector relation details view

### PR DESCRIPTION
**Pull request by anthonydillon**
The inspector now lists the relations and when a relation is clicked it shows the details for that relation.

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Load the GUI
- Add two relateable services to the canvas
- Make a relation between them
- Click on one of the services in the inspector
- Click on Relations
- Click on the new relation
- See that the relation information is displayed correctly
- Test it with multiple relations on a service
- Check the back from the relation details view goes back to the relations list
- Check it matches [the design](https://raw.githubusercontent.com/CanonicalLtd/juju-design/master/Inspector/Inspector_relations.png?token=ABWRnhlBz57vSzz3dZ8OxoYNmAx1ZdDcks5XMxoWwA%3D%3D)

File: jujugui/static/gui/src/app/components/deployment-bar/_deployment-bar.scss [hunk 1]
```diff
@@ -26,5 +26,10 @@
     .button--inline-neutral {
         padding-left: 10px;
         padding-right: 10px;
+        background-color: $canvas-background;
+
+        &:hover {
+          background-color: darken($canvas-background, 10%);
+        }
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/deployment-bar/_deployment-bar.scss]: Drive by so you can't see the service blocks behind the install button.

File: jujugui/static/gui/src/app/components/env-size-display/_env-size-display.scss [hunk 2]
```diff
@@ -21,7 +21,7 @@
         flex: 1;
         border: 1px solid $light-mid-grey;
         margin-bottom: 0;
-        background-color: transparent;
+        background-color: $canvas-background;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/env-size-display/_env-size-display.scss]: Drive by so you can't see the service blocks behind the unselected service/machine toggle buttons.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4990/
Test PASSed.

File: jujugui/static/gui/src/app/components/import-export/_import-export.scss [hunk 3]
```diff
@@ -12,6 +12,7 @@
     position: relative;
     top: 20px;
     transition: background-color .2s;
+    background-color: $canvas-background;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/import-export/_import-export.scss]: Drive by so you can't see the service blocks behind the import/export buttons.

File: jujugui/static/gui/src/app/components/import-export/_import-export.scss [hunk 4]
```diff
@@ -29,6 +30,8 @@
   }
 
   &__file {
+      position: absolute;
+      top: -99999;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/import-export/_import-export.scss]: Drive by to move the file input off screen so it doesnt block clicks in the header.

File: jujugui/static/gui/src/app/components/inspector-relation-details/inspector-relation-details.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('inspector-relation-details', function() {
+
+  juju.components.InspectorRelationDetails = React.createClass({
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector-relation-details/inspector-relation-details.js]: Will there be state and interaction methods for this component? If not, converting it to a stateless component might be a good idea.

File: jujugui/static/gui/src/app/components/inspector-relations/inspector-relations.js [hunk 6]
```diff
@@ -28,9 +28,16 @@ YUI.add('inspector-relations', function() {
     _generateRelations: function() {
       var relations = this.props.serviceRelations;
       var components = [];
-      relations.forEach(function(relation) {
+      if (relations.length === 0) {
+        return (
+          <div className="inspector-relations__message">
+            No relations for this service.
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector-relations/inspector-relations.js]: "There are no relations for this service yet. Click above to create one." ?

**Comment by hatched** [comment]
This looks great thanks, no qa yet.

File: jujugui/static/gui/src/app/components/inspector-relation-details/inspector-relation-details.js (hunk 5 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/inspector-relation-details/inspector-relation-details.js]: There will be state once I add build relation and delete relation. Is that what you mean't?

File: jujugui/static/gui/src/app/components/inspector-relations/inspector-relations.js (hunk 6 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/inspector-relations/inspector-relations.js]: I didn't want to add that yet as there is no where to click to add one. Will remember to update this message in the add-relation branch.

File: jujugui/static/gui/src/app/components/inspector-relation-details/inspector-relation-details.js (hunk 5 above)
Comment (hatched) [inline jujugui/static/gui/src/app/components/inspector-relation-details/inspector-relation-details.js]: I meant for this component specifically but yes I can see how that would prevent this being a stateless component. Thanks.

**Comment by hatched** [comment]
I like this new interaction, QA OK!

File: jujugui/static/gui/src/app/components/deployment-bar/_deployment-bar.scss [hunk 7]
```diff
@@ -26,5 +26,10 @@
     .button--inline-neutral {
         padding-left: 10px;
         padding-right: 10px;
+        background-color: $canvas-background;
+
+        &:hover {
+          background-color: darken($canvas-background, 10%);
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-bar/_deployment-bar.scss]: Missing a couple of spaces here.

File: jujugui/static/gui/src/app/components/inspector-relation-details/_inspector-relation-details.scss [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+    &__property {
+        @extend %stagger-fade-in;
+        margin-top: 0;
+        margin-bottom: .3em;
+    }
+
+    &__list {
+      list-style: none;
+      margin: 0;
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/inspector-relation-details/_inspector-relation-details.scss]: A couple more spaces here.

File: jujugui/static/gui/src/app/components/inspector-relations-item/inspector-relations-item.js [hunk 9]
```diff
@@ -20,70 +20,38 @@ YUI.add('inspector-relations-item', function() {
...
-         <span className="inspector-relations-item__service"
-          role="button" tabIndex="0"
-          onClick={this._handleServiceClick}>
-            <span className="inspector-relations-item__status">
-              <juju.components.SvgIcon name={this._getIcon(relation.pending)}
-                size="16" />
+        return (
+          <li className="inspector-relations-item">
+            <span className="inspector-relations-item__service"
+            onClick={this._handleRelationClick}
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/inspector-relations-item/inspector-relations-item.js]: These probably need to be indented a bit more.

File: jujugui/static/gui/src/app/components/inspector-relations-item/test-inspector-relations-item.js [hunk 10]
```diff
@@ -29,90 +29,39 @@ describe('InspectorRelationsItem', function() {
...
-            Scope: {"global"}
-          </p>
-        </span>
-      </li>);
+          relation={relation} />, true);
+    var instance = renderer.getMountedInstance();
+    var output = renderer.getRenderOutput();
+    var expected = (<li className="inspector-relations-item">
+      <span className="inspector-relations-item__service"
+      onClick={instance._handleRelationClick}
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/inspector-relations-item/test-inspector-relations-item.js]: A couple more spaces here too.

File: jujugui/static/gui/src/app/components/inspector/inspector.js [hunk 11]
```diff
@@ -269,6 +269,25 @@ YUI.add('inspector-component', function() {
                   activeComponent: undefined
                 }}}};
           break;
+        case 'relation':
+          var relationIndex = metadata.relation;
+          var relation = this.props.serviceRelations[relationIndex];
+          var serviceName = relation.far.serviceName;
+          var relationName =relation.far.name;
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/inspector/inspector.js]: Space after the =

**Comment by huwshimi** [comment]
Great stuff @anthonydillon  :+1: with a few trivial whitespace changes.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4998/
Test PASSed.

**Comment by anthonydillon** [comment]
Thanks @hatched @huwshimi :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**